File: Array/linchuanxin/Algorithm/day30_206. 反转链表/day30_leetcode206.kt
package day30


  class ListNode(var `val`: Int) {
      var next: ListNode? = null
  }
class Solution {
    fun reverseList(head: ListNode?): ListNode? {
        var pre:ListNode? = null
        var cur:ListNode? = head

        while (cur!=null){
            var tmp = cur.next
            cur.next = pre
            pre = cur
            cur = tmp
        }

        return pre

    }
}

File: Array/linchuanxin/Algorithm/day30_0630_206. 反转链表/day30_leetcode206.c
day30_leetcode_206//
// Created by LCX on 2020/7/1.
//
#include <stdio.h>


struct ListNode {
    int val;
    struct ListNode *next;
};



struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *pre = NULL;
    struct ListNode *cur = head;

    while (cur != NULL)
    {
        struct ListNode *tmp = cur->next;
        cur->next = pre;
        pre = cur;
        cur = tmp;
    }

    return pre;
}